app/models/binary/explore/PrecomputedExplorer.scala

```diff
@@ -0,0 +1,64 @@
+import com.scalableminds.webknossos.datastore.datareaders.precomputed.{PrecomputedHeader, PrecomputedScale}
+import com.scalableminds.webknossos.datastore.models.datasource.{Category, ElementClass}
+
+import java.nio.file.Path
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class PrecomputedExplorer extends RemoteLayerExplorer {
+  override def name: String = "Neuroglancer Precomputed"
+
+  override def explore(remotePath: Path, credentialId: Option[String]): Fox[List[(PrecomputedLayer, Vec3Double)]] =
+    for {
+      infoPath <- Fox.successful(remotePath.resolve(PrecomputedHeader.METADATA_PATH))
+      precomputedHeader <- parseJsonFromPath[PrecomputedHeader](infoPath) ?~> s"Failed to read Precomputed metadata at $infoPath"
+      layerAndVoxelSize <- layerFromPrecomputedHeader(precomputedHeader, remotePath, credentialId)
+    } yield List(layerAndVoxelSize)
+
+  private def layerFromPrecomputedHeader(precomputedHeader: PrecomputedHeader,
+                                         remotePath: Path,
+                                         credentialId: Option[String]): Fox[(PrecomputedLayer, Vec3Double)] =
+    for {
+      name <- guessNameFromPath(remotePath)
+      boundingBox <- BoundingBox.fromSize(precomputedHeader.scales.head.size).toFox
+      elementClass: ElementClass.Value <- elementClassFromPrecomputedDataType(precomputedHeader.data_type) ?~> "Unknown data type"
+      smallestResolution = precomputedHeader.scales.head.resolution
+      voxelSize <- Vec3Int.fromList(smallestResolution.toList).toFox
+      mags: Seq[MagLocator] <- Fox.serialCombined(precomputedHeader.scales)(
+        getMagFromScale(_, smallestResolution, remotePath, credentialId))
+      layer = if (precomputedHeader.describesSegmentationLayer) {
+        PrecomputedSegmentationLayer(name, boundingBox, elementClass, mags.toList, None)
+      } else PrecomputedDataLayer(name, boundingBox, Category.color, elementClass, mags.toList)
+    } yield (layer, Vec3Double.fromVec3Int(voxelSize))
+
+  private def elementClassFromPrecomputedDataType(precomputedDataType: String): Fox[ElementClass.Value] =
+    precomputedDataType.toLowerCase match {
+      case "uint8"   => Some(ElementClass.uint8)
+      case "uint16"  => Some(ElementClass.uint16)
+      case "uint32"  => Some(ElementClass.uint32)
+      case "uint64"  => Some(ElementClass.uint64)
+      case "float32" => Some(ElementClass.float)
+      case _         => None
+    }
+
+  private def getMagFromScale(scale: PrecomputedScale,
+                              minimalResolution: Array[Int],
+                              remotePath: Path,
+                              credentialId: Option[String]): Fox[MagLocator] = {
+    val normalizedResolution = (scale.resolution, minimalResolution).zipped.map((r, m) => r / m)
+    for {
+      mag <- Vec3Int.fromList(normalizedResolution.toList)
+      path = remotePath.resolve(scale.key)
+      axisOrder = AxisOrder(0, 1, 2)
```

Comment:
Do we know that this is always the case? Is this specific axis order known only from experiments? Maybe you can find a part in the spec where this is described? If so, this would be a good spot for a link in a comment

Reply:
Not really described that well, but https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/volume.md#unsharded-chunk-storage lays out how chunks are indexed, and explicitly writes x then y then z.

Reply:
Yes that does the trick, could you add a comment to the AxisOrder line noting that, and possibly including the link?